toolkit/tools/pkg/imagecreatorlib/validation.go

```diff
@@ -0,0 +1,127 @@
+package imagecreatorlib
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/azurelinux/toolkit/tools/pkg/imagecustomizerlib"
+)
+
+func validateSupportedFields(c *imagecustomizerapi.Config) error {
+	// Verify that the config file does not contain any fields that are not supported by the image creator.
+	if c.Input != (imagecustomizerapi.Input{}) {
+		return fmt.Errorf("input field is not supported by the image creator")
+	}
+	if c.Iso != nil {
+		return fmt.Errorf("iso field is not supported by the image creator")
+	}
+	if c.Pxe != nil {
+		return fmt.Errorf("pxe field is not supported by the image creator")
+	}
+
+	if len(c.PreviewFeatures) > 0 {
+		return fmt.Errorf("preview features field is not supported by the image creator")
+	}
+
+	if c.Storage.ResetPartitionsUuidsType != imagecustomizerapi.ResetPartitionsUuidsTypeDefault {
+		return fmt.Errorf("reset partitions uuids field is not supported by the image creator")
+	}
+
+	if c.Storage.Verity != nil {
+		return fmt.Errorf("storage verity field is not supported by the image creator")
+	}
+
+	if c.OS != nil && len(c.OS.AdditionalFiles) > 0 {
```

Comment:
It might be good to use a helper function for all the OS fields to avoid duplicating the `c.OS != nil` check.